src/ui/public/vislib/visualizations/area_chart.js

```diff
@@ -165,25 +134,21 @@ export default function AreaChartFactory(Private) {
      */
     addCircles(svg, data) {
       const color = this.handler.data.getColorFunc();
-      const xScale = this.handler.categoryAxes[0].getScale();
-      const yScale = this.handler.valueAxes[0].getScale();
+      const xScale = this.getCategoryAxis().getScale();
+      const yScale = this.getValueAxis().getScale();
       const ordered = this.handler.data.get('ordered');
       const circleRadius = 12;
       const circleStrokeWidth = 0;
       const tooltip = this.tooltip;
       const isTooltip = this._attr.addTooltip;
       const isOverlapping = this.isOverlapping;
 
-      const layer = svg.selectAll('.points')
-      .data(data)
-      .enter()
-      .append('g')
+      const layer = svg.append('g')
       .attr('class', 'points area');
 
       // append the circles
-      const circles = layer
-      .selectAll('circles')
-      .data(function appendData(data) {
+      const circles = layer.selectAll('circles')
+      .data(function appendData() {
         return data.filter(function isZeroOrNull(d) {
```

Comment:
oh, before data was added to html elements (d3), now we just pass it in to the function.